Documentation/Contributors/CodingGuide/README.md

```diff
@@ -0,0 +1,660 @@
+var f = 1.0;
+```
+* Declare variables where they are first used.  For example,
+```javascript
+var i;
+var m;
+var models = [ /* ... */ ];
+var length = models.length;
+for (i = 0; i < length; ++i) {
+    m = models[i];
+    // Use m
+}
+```
+is better written as
+```javascript
+var models = [ /* ... */ ];
+var length = models.length;
+for (var i = 0; i < length; ++i) {
+    var m = models[i];
+    // Use m
+}
+```
+* Variables have function-level, not block-level scope.  Do not rely on variable hoisting, i.e., using a variable before it is declared, e.g.,
+```javascript
+console.log(i); // i is undefined here.  Never use a variable before it is declared.
+var i = 0.0;
+```
+* Use `undefined` instead of `null`.
+* Test if a variable is defined using Cesium's `defined` function, e.g.,
+```javascript
+var v = undefined;
+if (defined(v)) {
+    // false
+}
+
+var u = {};
+if (defined(u)) {
+    // true
+}
+```
+* Use Cesium's `definedNotNull` function to test the return of `String.match()` and JSON properties.
+* Use Cesium's `freezeObject` function to create enums, e.g.,
+```javascript
+/*global define*/
+define([
+        '../Core/freezeObject'
+    ], function(
+        freezeObject) {
+    "use strict";
+
+    return freezeObject({
+        STOPPED : 0,
+        ANIMATING : 1
+    });
+});
+```
+* Use descriptive comments for non-obvious code, e.g.,
+```javascript
+byteOffset += sizeOfUint32; // Add 4 to byteOffset
+```
+is better written as
+```javascript
+byteOffset += sizeOfUint32; // Skip length field
+```
+* `TODO` comments need to be removed or addressed before the code is merged into master.  Used sparingly, `PERFORMANCE_IDEA`, can be handy later when profiling.
+
+## Functions
+
+* :art: Functions should be **cohesive**; they should only do one task.
+* Statements in a function should be at a similar level of abstraction.  If a code block is much lower level than the rest of the statements, it is a good candidate to move to a helper function, e.g., 
+```javascript
+Cesium3DTileset.prototype.update = function(frameState) {
+    var tiles = this._processingQueue;
+    var length = tiles.length;
+
+    for (var i = length - 1; i >= 0; --i) {
+        tiles[i].process(this, frameState);
+    }
+
+    selectTiles(this, frameState);
+    updateTiles(this, frameState);
+};
+```
+is better written as
+```javascript
+Cesium3DTileset.prototype.update = function(frameState) {
+    processTiles(this, frameState);
+    selectTiles(this, frameState);
+    updateTiles(this, frameState);
+};
+
+function processTiles(tiles3D, frameState) {
+    var tiles = tiles3D._processingQueue;
+    var length = tiles.length;
+
+    for (var i = length - 1; i >= 0; --i) {
+        tiles[i].process(tiles3D, frameState);
+    }
+}
+```
+* :speedboat: Smaller functions are more likely to be optimized by V8.  Consider this for code that is likely to be a hot spot.
+* :speedboat: Avoid multiple `return` statements in small hot functions since V8 will not inline them .
```

Comment:
http://nodeup.com/fortyfive

I'm fine with not including it if we think otherwise.